### PR TITLE
[DR-80433] Replace moment.js with date-fns in shared folder

### PR DESCRIPTION
**Pull request by anniebtran**
## Summary

Since moment.js is no longer supported, we are going through and removing where we're using moment.js and using date-fns instead, beginning with our shared folder — there was only 1 shared function to replace (`getDate`) but it's called all across our forms/tests so we touch quite a few files here.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/80433

## Testing done
- [x] Added unit tests for the new shared function that uses `date-fns`
- [x] Existing unit and E2E tests pass where we've substituted the new `date-fns` based function

## What areas of the site does it impact?

Appeals (Decision Reviews)

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

@Mottie I left a few comments with questions that I'm hoping you can answer since you're more familiar with this code. Also let me know if there's any test coverage I need to add — it seemed like we had pretty good coverage, but I may have missed some opportunities to add some date specific tests

### CODE REVIEW

File: src/applications/appeals/shared/utils/issues.jsx [hunk 1]
```diff
@@ -17,6 +25,13 @@ export const getIssueName = (entry = {}) =>
 export const getIssueDate = (entry = {}) =>
   entry.decisionDate || entry.attributes?.approxDecisionDate || '';
 
+export const getDecisionDate = issue => {
+  const dateToParse =
+    issue.attributes?.approxDecisionDate || issue.decisionDate || ''; // any reason not to use `getIssueDate` above? the ordering is different than this one so wasn't sure
```
Comment (anniebtran) [inline src/applications/appeals/shared/utils/issues.jsx:30]: cc: @Mottie

File: src/applications/appeals/995/tests/utils/helpers.unit.spec.js [hunk 2]
```diff
@@ -24,21 +24,21 @@ import {
 } from '../../../shared/utils/issues';
 
 describe('getEligibleContestableIssues', () => {
-  const date = moment().startOf('day');
+  const date = startOfDay(new Date());
```
Comment (Mottie) [inline src/applications/appeals/995/tests/utils/helpers.unit.spec.js:27]: I was exploring date-fns today and found a `startOfToday` function :D - no need to update this though

**Review by Mottie — CHANGES_REQUESTED** [review]

File: src/applications/appeals/995/components/EvidenceSummaryLists.jsx [hunk 3]
```diff
@@ -42,7 +46,10 @@ const removeButtonClass = [
 ].join(' ');
 
 const formatDate = (date = '') => {
-  const result = getDate({ date, pattern: FORMAT_COMPACT });
+  // Use `parse` from date-fns because it is a non-ISO8061 formatted date string
+  const parsedDate = parse(date, FORMAT_YMD_DATE_FNS, new Date());
```
[the remark this answers is not in the document]
Comment (anniebtran) [reply src/applications/appeals/995/components/EvidenceSummaryLists.jsx:50]: Hmmm, I can give it a try

**Review by Mottie — APPROVED** [review]
LGTM

**Pull request merged**